### PR TITLE
Remove sporadic min/max usage estimates from stats

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today `GET _nodes/stats/fs` includes `{least,most}_usage_estimate`
fields for some nodes. These fields have rather strange semantics. They
are only reported on the elected master and on nodes that have been the
elected master since they were last restarted; when a node stops being
the elected master these stats remain in place but we stop updating them
so they may become arbitrarily stale.

This means that these statistics are pretty meaningless and impossible
to use correctly. Even if they were kept up to date they're never
reported for data-only nodes anyway, despite the fact that data nodes
are the ones where we care most about disk usage. The information needed
to compute the path with the least/most available space is already
provided in the rest the stats output, so we can treat the inclusion of
these stats as a bug and fix it by simply removing them in this commit.
Since these stats were always optional and mostly omitted (for opaque
reasons) this is not considered a breaking change.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/monitor/fs/FsInfo.java [hunk 1]
```diff
@@ -119,19 +120,6 @@ private long addLong(long current, long other) {
             return current + other;
         }
 
-        private double addDouble(double current, double other) {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/monitor/fs/FsInfo.java:122]: Unrelated unused method

File: server/src/main/java/org/elasticsearch/monitor/fs/FsService.java [hunk 2]
```diff
@@ -49,23 +44,22 @@
...
         logger.debug("using refresh_interval [{}]", refreshInterval);
-        cache = new FsInfoCache(refreshInterval, stats(probe, null, logger, null));
+        cache = new FsInfoCache(refreshInterval, stats(probe, null));
     }
 
     public FsInfo stats() {
         return cache.getOrRefresh();
     }
 
-    private static FsInfo stats(FsProbe probe, FsInfo initialValue, Logger logger, @Nullable ClusterInfo clusterInfo) {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/monitor/fs/FsService.java:64]: Unrelated but `logger` was always the static field so no need to pass it in.

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
Thanks Yannick

**Pull request merged**